src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -875,7 +875,7 @@ def makeSubmit(self, jobList):
             jdl.append("+JobLeaseDuration = isUndefined(MachineAttrMaxHibernateTime0) ? 1200 : MachineAttrMaxHibernateTime0\n")
 
             ### print all the variables needed for us to rely on condor userlog
-            jdl.append("job_ad_information_attrs = JobStatus,QDate,EnteredCurrentStatus,JobStartDate,DESIRED_Sites,ExtDESIRED_Sites,WMAgent_JobID,MATCH_EXP_JOBGLIDEIN_CMSSite\n")
+            jdl.append("job_ad_information_attrs = JobStatus,QDate,EnteredCurrentStatus,DESIRED_Sites,ExtDESIRED_Sites,WMAgent_JobID,MATCH_EXP_JOBGLIDEIN_CMSSite\n")
```

Comment:
Yeah - that's precisely why the original log code doesn't work.

The _schedd_ evaluates the expression and saves the resulting string as `MATCH_EXP_JOBGLIDEIN_CMSSite`.  That's what you see in the output of `condor_q` or `condor_history`.  However, when it sends the ClassAd to the `condor_shadow` (the `condor_shadow` writes the log), it saves the resulting string to `JOBGLIDEIN_CMSSite`.

Hence the shadow, when writing the log, sees something like `T2_US_Nebraska` instead of `$$([ifThenElse(...)])`.

(If you can't tell, this isn't the recommended way to get this information into the log - it's a bit roundabout and seems prone to issues.)